### PR TITLE
Getting started with Spring Boot REST Template

### LINKED ISSUE #2106: Getting started with Spring Boot Rest Template
Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article on time from inception to publishing.)**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Getting started with Spring Boot RestTemplate
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article's higher ranking.*

### Introduction paragraph (2-3 paragraphs):
Interacting with REST APIs in Java requires a lot of boilerplate code that makes the final application code complex. RestTemplate makes interacting with most RESTful web services a one-line incantation. RestTemplate simplifies REST API interaction in Spring Boot applications by abstracting away the boilerplate code that would slow down the development process and providing a single-line implementation.  In the tutorial, I will be explaining how to consume RESTful APIs from the popular [JSON placeholder](https://jsonplaceholder.typicode.com/).

### Key takeaways: 
1. Setting up RestTemplate in Spring Boot.
2. Consuming data from the various REST API endpoints using RestTemplate.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by elizabeth962**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2106

### CODE REVIEW

File: content/articles/spring-boot-rest-template/index.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+**PUT request**
+![PUT request](put-request.png)
+
+**DELETE request**
+![DELETE request](delete-request.png)
+
+
+### Conclusion
+Now that you have learned how to consume RESTful web services through Spring Boot Rest template, create a Spring Boot application that exposes its services through REST endpoints and Consume the endpoints from another Spring Boot application. Find the complete source code for the application [here]().
+
```
Comment (paulodhiambo) [inline content/articles/spring-boot-rest-template/index.md:187]: Add the link to complete source code

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@elizabeth962 add your author profile and review the requested changes. Ping me when done.

**Comment by paulodhiambo** [comment]
@elizabeth962 good tutorial. @hectorkambow you can give it a final review.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
great work @elizabeth962 

Looking forward to more amazing articles 🚀 

changed the hero just because I notice a minor misspelling - we can replace at any given time.

cc @paulodhiambo

**Pull request merged**